common/src/chain/transaction/output/mod.rs

```diff
@@ -39,6 +39,15 @@ pub mod output_value;
 pub mod stakelock;
 pub mod timelock;
 
+fixed_hash::construct_fixed_hash! {
+    #[derive(Encode, Decode, serde::Serialize, serde::Deserialize)]
```

Comment:
It's unacceptable to have both Encode and Serialize together. Same for Decode and Deserialize. We have learned from recent events that this can only lead to problems serde_json representation.

For RPC, you can create special wrappers for this.